ovos_bus_client/users.py

```diff
@@ -0,0 +1,223 @@
+import enum
+import time
+from threading import Lock
+from uuid import uuid4
+
+from ovos_config.config import Configuration
+from ovos_config.locale import get_default_tz
+from ovos_utils.log import LOG
+
+from ovos_bus_client.message import dig_for_message, Message
+from ovos_bus_client.session import SessionManager, Session
+from ovos_backend_client.api import GeolocationApi
+
+
+def _config2preferences(cfg=None):
+    # TODO - filter relevant keys only, ensure no sensistive or irrelevant info passed along
+    if cfg is None:  # allow empty dict to mean "no preferences"
+        cfg = dict(Configuration.get())
+    return cfg or {}
+
+
+def _config2location(cfg=None):
+    loc = cfg or Configuration.get()["location"]
+    if isinstance(loc, dict):
+        loc = Location.from_dict(cfg)
+    assert isinstance(loc, Location)
+    return loc
+
+
+class Location:
+    geolocation = GeolocationApi()
+
+    def __init__(self, lat, lon):
+        self.latitude = lat
+        self.longitude = lon
+        # everything is automatically derived from lat/lon
+        # this ensures location is not ambiguous
+        self._address = None
+        self._timezone_code = None
+        self._city = None
+        self._state = None
+        self._country = None
+        self._country_code = None
+
+    @property
+    def timezone_code(self):
+        if self._timezone_code is None:
+            self._timezone_code = get_timezone(self.latitude, self.longitude)
+        return self._timezone_code
+
+    @property
+    def timezone(self):
+        if self.timezone_code:
+            return gettz(self.timezone_code)
+        return get_default_tz()
+
+    @property
+    def current_time(self):
+        return datetime.datetime.now(self.timezone)
+
+    @property
+    def address(self):
```

Comment:
Should this return an actual street address if one is available? Seems it will just be a formatted string in any case, so more for user consumption than code

Reply:
this is just user facing friendly string, not consumed anywhere, meant to use in things like GUI if a display friendly name is needed. could also be used for generating TTS i suppose